backends-velox/src/main/scala/io/glutenproject/backendsapi/velox/VeloxBackend.scala

```diff
@@ -53,9 +56,10 @@ object VeloxBackend {
   val BACKEND_NAME = "velox"
 }
 
-object BackendSettings extends BackendSettingsApi {
+object BackendSettings extends BackendSettingsApi with Logging {
 
   val SHUFFLE_SUPPORTED_CODEC = Set("lz4", "zstd")
+  val SUPPORTED_FILE_SYSTEM = Set("file", "hdfs", "s3a")
```

Comment:
oss can also be supported via s3a scheme.

Reply:
Maybe, we should include all schemes in the below link and name this set `SUPPORTED_SCHEMES`. @leesf, does it make sense?
https://github.com/facebookincubator/velox/blob/aaaa0791008267d8be0a5c141db6da261aba8892/velox/connectors/hive/storage_adapters/s3fs/S3Util.h#L34-L45

Reply:
If it is named `SUPPORTED_SCHEMES`, maybe each scheme name should be xxx://.

Reply:
how about we provide a blacklist configuration to exclude?

Reply:
all schemes:

```
Set(
 "file", // local
 "hdfs", // hdfs
 "s3", "s3a", "s3n", "oss", "oss", "cosn",  // s3
 "gs", // gcs
 "abfs", "abfss"  // abfs
) 
```

Reply:
> If it is named `SUPPORTED_SCHEMES`, maybe each scheme name should be xxx://.

Do we need that? The scheme obtained by `java.net.URI#getScheme` does not have the '://' suffix.

Reply:
Maybe, using a configurable blacklist is not necessary.

BTW, I note velox registers S3/HDFS/GCS/ABFS according to the corresponding build option. If the validation passes on Gluten side, but velox doesn't actually register the file system. Can the expected fallback still happen?
https://github.com/facebookincubator/velox/blob/aaaa0791008267d8be0a5c141db6da261aba8892/velox/connectors/hive/HiveConnector.cpp#L151-L159

Reply:
> > If it is named `SUPPORTED_SCHEMES`, maybe each scheme name should be xxx://.
> 
> Do we need that? The scheme obtained by `java.net.URI#getScheme` does not have the '://' suffix.

OK to me if "://" is lacking.

Reply:
> BTW, I note velox registers S3/HDFS/GCS/ABFS according to the corresponding build option. If the validation passes on Gluten side, but velox doesn't actually register the file system. Can the expected fallback still happen?

Good point. How about we add relevant comments in the code? It prompts to recompile velox for support.

Reply:
OK. Please add some comments to clarify. Thanks!

Reply:
> OK. Please add some comments to clarify. Thanks!

Added, please help to continue the review, thanks.

Reply:
@PHILO-HE agree that we should not break the capabilities velox already have. In addition, if users add new schemes to velox(which is compatible with s3a), we must also fix it in gluten, which would make it a little complex. But I am open with PR.

Reply:
>  if users add new schemes to velox(which is compatible with s3a), we must also fix it in gluten

For making file system support state always consistent with velox, one way is to do the validation on native side, i.e., calling velox [getFileSystem](https://github.com/facebookincubator/velox/blob/5144f141cfe2cc832d324085ed570457c22a9a77/velox/common/file/FileSystems.cpp#L52) to validate. @wForget, would you like to improve this PR?

Reply:
Sure, I will try it.

Reply:
> > if users add new schemes to velox(which is compatible with s3a), we must also fix it in gluten
> 
> For making file system support state always consistent with velox, one way is to do the validation on native side, i.e., calling velox [getFileSystem](https://github.com/facebookincubator/velox/blob/5144f141cfe2cc832d324085ed570457c22a9a77/velox/common/file/FileSystems.cpp#L52) to validate. @wForget, would you like to improve this PR?

@PHILO-HE  I added a native method to check paths, could you please help me take a look?